packages/authenticator/amplify_authenticator/lib/src/blocs/auth/auth_bloc.dart

```diff
@@ -227,10 +227,14 @@ class StateMachineBloc
           yield UnauthenticatedState.confirmSignInNewPassword;
         case AuthSignInStep.confirmSignInWithTotpMfaCode:
           yield UnauthenticatedState.confirmSignInWithTotpMfaCode;
+        case AuthSignInStep.confirmSignInWithEmailMfaCode:
+          yield UnauthenticatedState.confirmSignInWithEmailMfaCode;
         case AuthSignInStep.continueSignInWithMfaSelection:
           yield ContinueSignInWithMfaSelection(
             allowedMfaTypes: result.nextStep.allowedMfaTypes,
           );
+        case AuthSignInStep.continueSignInWithMfaSetupSelection:
+          await _handleMfaSetupSelection(result);
```

Comment:
small-refactor: `_handleMfaSetupSelection` should return the new state and yielded it here to avoid unexpected behavior with the stream.